nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-processors/src/main/java/org/apache/nifi/processors/mongodb/AbstractMongoProcessor.java

```diff
@@ -159,11 +161,42 @@ public abstract class AbstractMongoProcessor extends AbstractProcessor {
     static final List<PropertyDescriptor> descriptors;
 
     static {
-        List<PropertyDescriptor> _temp = new ArrayList<>();
-        _temp.add(CLIENT_SERVICE);
-        _temp.add(DATABASE_NAME);
-        _temp.add(COLLECTION_NAME);
-        descriptors = Collections.unmodifiableList(_temp);
+      descriptors = List.of(CLIENT_SERVICE, DATABASE_NAME, COLLECTION_NAME);
+    }
+
+    public enum MongoUpdateOption implements DescribedValue {
```

Comment:
As this is an `enum` under the Processor, I believe the name can be shortened. I recommend using `UpdateMethod` to match the new property name, even though the property is still named `Update Mode` in the `PutMongoRecord` Processor.
```suggestion
    public enum UpdateMethod implements DescribedValue {
```